lib/chef/knife/ssh.rb

```diff
@@ -160,6 +160,31 @@ def configure_session
         session_from_list(list)
       end
 
+      def get_ssh_attribute(node)
+        # Order of precedence for ssh target
+        # 1) command line attribute
+        # 2) configuration file
+        # 3) cloud attribute
+        # 4) fqdn
+        if config[:attribute_from_cli]
```

Comment:
It's been a while since I looked at this, but, ```config[:attribute_from_cli]``` was something we created inside #configure_attribute to preserve before we mangled ```config[:attribute]```. You've since removed #configure_attribute, so this would never be set. Maybe you just want to change this to ```config[:attribute]```?